extensions/rich_text_components/components.py

```diff
@@ -0,0 +1,172 @@
+# coding: utf-8
+#
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Classes for Rich Text Components in Oppia."""
+
+import re
+
+import bs4
+from extensions.objects.models import objects
+
+
+class BaseRteComponent(object):
+    """Base object class.
+
+    This is the superclass for rich text components in Oppia, such as
+    Image and Video.
+    """
+
+    # These values should be overridden in subclasses.
+    customization_args = {}
```

Comment:
Make this and arg_schema None. Add a backend test to ensure that these values are non-None for all individual components.

Add a single-line docstring above each of these properties explaining what they are. (For example I don't really know what arg schema is. What's an "arg" in the context of an HTML component? You probably want attr_value_schemas or similar.)

Reply:
Removed `arg_schema`, only a single source for `arg_names_to_obj_classes` now.